src/keyboard.rs

```diff
@@ -833,6 +833,39 @@ pub fn translate_key_code(peer: &str, event: &Event, key_event: KeyEvent) -> Opt
     map_keyboard_mode(peer, event, key_event)
 }
 
+#[inline]
+fn is_altgr(event: &Event) -> bool {
+    #[cfg(target_os = "linux")]
+    if event.platform_code == 0xFE03 {
+        true
+    } else {
+        false
+    }
+
+    #[cfg(target_os = "macos")]
+    // ignore right option key
+    if event.platform_code as u32 == rdev::kVK_RightOption {
+        true
+    } else {
+        false
+    }
+
+    #[cfg(target_os = "windows")]
+    if unsafe { IS_0X021D_DOWN } {
```

Comment:
`if unsafe { IS_0X021D_DOWN } && event.position_code == 0xE038` may be more clear.